### PR TITLE
SALTO-2114 app installation will lose the "settings_object" item

### PR DESCRIPTION
**Pull request by yanaise**
App installation will no longer include unnecessary "settings_object" attribute, which is a duplicate of "settings" attribute

---

---
_Release Notes_: 
Zendesk:
App installation will no longer include unnecessary "settings_object" attribute, which is a duplicate of "settings" attribute
---
_User Notifications_:
Zendesk: 
Future fetches of zendesk will no longer include unnecessary "settings_object" attribute, which is a duplicate of "settings" attribute. Current workspaces will continue to work with no need to change.

### CODE REVIEW

File: packages/zendesk-support-adapter/src/filters/handle_template_expressions.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ type TemplateContainer = {
   template: PotentialTemplateField
 }
 
-const zendeskReferenceTypeToSaltoType: Record<string, string> = {
+export const zendeskReferenceTypeToSaltoType: Record<string, string> = {
```
Comment (noamhammer) [inline packages/zendesk-support-adapter/src/filters/handle_template_expressions.ts:45]: make sure if there can be references to user_field and organization_field

File: packages/zendesk-support-adapter/src/filters/handle_app_installations.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        ), e => e.elemID.typeName))),
+    'Create template creation filter'),
+    preDeploy: (changes: Change<InstanceElement>[]): Promise<void> => log.time(async () => {
+      getAppInstallations(changes.map(getChangeData).filter(isInstanceElement))
+        .forEach(app => {
+          const convertTemplatesToValues = (field: string): void => {
+            if (isTemplateExpression(app.value.settings[field])) {
+              const templateUsingIdField = createTemplateUsingIdField(app.value.settings[field],
+                true)
+              deployTemplateMapping[templateUsingIdField.value] = app.value.settings[field]
```
Comment (noamhammer) [inline packages/zendesk-support-adapter/src/filters/handle_app_installations.ts:119]: add safties

File: packages/zendesk-support-adapter/src/filters/handle_app_installations.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            }
+          }
+          runFunctionOnLocatedFields(app, convertTemplatesToValues)
+        })
+    }, 'Create template resolve filter'),
+    onDeploy: async (changes: Change<InstanceElement>[]): Promise<void> => log.time(async () =>
+      getAppInstallations(changes.map(getChangeData).filter(isInstanceElement))
+        .forEach(app => {
+          const resolveTemplates = (field: string): void => {
+            app.value.settings[field] = deployTemplateMapping[app.value.settings[field]]
```
Comment (noamhammer) [inline packages/zendesk-support-adapter/src/filters/handle_app_installations.ts:130]: add safties

File: packages/zendesk-support-adapter/src/filters/app.ts [hunk 4]
```diff
@@ -83,6 +84,12 @@ Promise<void> => {
  * Deploys app installation
  */
 const filterCreator: FilterCreator = ({ config, client }) => ({
+  onFetch: async (elements: Element[]) =>
```
[the remark this answers is not in the document]
Comment (yanaise) [reply packages/zendesk-support-adapter/src/filters/app.ts:87]: yeah, I was always planning to, just wanted to make sure I did i in the way you wanted

File: packages/zendesk-support-adapter/src/filters/handle_app_installations.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+const locators: IdLocator[] = [{
+  fieldRegex: /.*_fields/g,
+  idRegex: /(?:([\d]+)\s*,\s*|([\d]+)$)/g,
+  type: 'ticket_field',
+}]
+
+const valueToTemplate = (val: string, regex: RegExp, locatorType: string,
+  typeToElements: Record<string, InstanceElement[]>): TemplateExpression | string => {
+  const matches = Array.from(matchAll(val, regex))
+    .flatMap(match => Array.from(match.slice(1))).filter(e => !_.isUndefined(e))
```
[the remark this answers is not in the document]
Comment (yanaise) [reply packages/zendesk-support-adapter/src/filters/handle_app_installations.ts]: match[0] is the whole expression, rather than groups

**Pull request merged**